eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,42 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper([string]$subscription, [string]$resourceGroup, [string]$displayName)
+{
+    $servicePrincipal = Retry {
+        New-AzADServicePrincipal -Role "Owner" -Scope "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName" -DisplayName $displayName
+    }
+    $spPassword = ""
+    $appId = ""
```

Comment:
Tip: you don't need to do this. Setting these variables in the following conditional blocks are within the same scope. Only new script blocks (like passed to a `foreach-object`) constitute new scopes.

Reply:
I know, but I think it can cause issues later on when refactoring and removing blocks that may or may not have variable definition dependencies. I try to declare all variables used above to defend against this type of issue and make it clear what variables will get defined and used in scope.